motor/discovery/aws/ec2_instances.go

```diff
@@ -146,33 +158,53 @@ func (ec2i *Ec2Instances) List() ([]*asset.Asset, error) {
 	return instances, nil
 }
 
-func instanceToAsset(account string, region string, instance types.Instance, insecure bool, passInLabels map[string]string) *asset.Asset {
+type instanceInfo struct {
+	account      string
+	region       string
+	instance     types.Instance
+	insecure     bool
+	passInLabels map[string]string
+	profile      string
+	imageName    string
+}
+
+func instanceToAsset(instanceInfo instanceInfo) *asset.Asset {
 	asset := &asset.Asset{
-		PlatformIds: []string{awsec2.MondooInstanceID(account, region, *instance.InstanceId)},
+		PlatformIds: []string{awsec2.MondooInstanceID(instanceInfo.account, instanceInfo.region, *instanceInfo.instance.InstanceId)},
 		Connections: []*providers.Config{},
 		Labels:      make(map[string]string),
 		IdDetector:  []string{"awsec2"},
-		Name:        *instance.InstanceId,
+		Name:        *instanceInfo.instance.InstanceId,
 		Platform: &platform.Platform{
 			Kind:    providers.Kind_KIND_VIRTUAL_MACHINE,
 			Runtime: providers.RUNTIME_AWS_EC2,
 		},
-		State: mapEc2InstanceStateCode(instance.State),
+		State: mapEc2InstanceStateCode(instanceInfo.instance.State),
 	}
 
 	// if there is a public ip, we assume ssh is an option
-	if instance.PublicIpAddress != nil {
+	if instanceInfo.instance.PublicIpAddress != nil {
```

Comment:
Great call. Could we further improve it by checking via the API if the ssm agent is installed and or the permissions are set?

Reply:
we get the ssm info during the ssm part of the discovery and then merge them

not sure about checking if the permissions are set, i'll have to look into that 🤔